components/test_pd/src/mocker/mod.rs

```diff
@@ -216,4 +220,11 @@ pub trait PdMocker {
     fn get_operator(&self, _: &GetOperatorRequest) -> Option<Result<GetOperatorResponse>> {
         None
     }
+
+    fn upload_ru_metrics(&self, req: &TokenBucketsRequest) -> Option<Result<TokenBucketsResponse>> {
+        req.get_requests().iter().for_each(|r| {
+            assert_eq!(r.get_is_background(), true);
```

Comment:
@glorv FIY, just check the label for req...

Reply:
But how can you ensure if this method is even called?

Reply:
- Change `upload_duration` to 10ms in cfg and wait 100 ms here to make sure it does.
- The `new_test_server_and_client` ensures that we pass the simulation pd.